pathfinder/pathfinding_service.py

```diff
@@ -234,6 +236,34 @@ def on_balance_proof_message(self, balance_proof: BalanceProof):
                 balance_proof.locked_amount,
             )
 
+    def on_paths_request_message(self, paths_request: PathsRequest):
+        self._check_chain_id(paths_request.chain_id)
+
+        token_network = self._get_token_network(paths_request.token_network_address)
+
+        if token_network:
+            log.debug('Received PathsRequest message for token network {}'.format(
+                token_network.address
+            ))
+            paths_reply: PathsReply = PathsReply(
+                paths_request.token_network_address,
+                paths_request.target_address,
+                paths_request.value,
+                paths_request.chain_id,
+                nonce=1,
+                paths_and_fees=token_network.get_paths(
+                        paths_request.source_address,
+                        paths_request.target_address,
+                        paths_request.value,
+                        paths_request.num_paths,
+
+                ),
+                signature=''
+
+            )
+            # FIXME: This method should send a reply message back to the requesting client
+            return paths_reply
```

Comment:
Here we need a `send` method on the `Transport`. I think we should add that to the `Transport` now, even though it will probably change. But then the core functionality is implemented correctly.